rust/main/chains/hyperlane-cosmos/src/cw/mailbox/dispatch_indexer.rs

```diff
@@ -9,42 +10,47 @@ use tracing::instrument;
 
 use hyperlane_core::{
     ChainCommunicationError, ChainResult, ContractLocator, Decode, HyperlaneMessage, Indexed,
-    Indexer, LogMeta, SequenceAwareIndexer, H512,
+    Indexer, LogMeta, SequenceAwareIndexer, H256, H512,
 };
 
-use crate::rpc::{CosmosWasmRpcProvider, ParsedEvent, WasmRpcProvider};
-use crate::utils::{
-    execute_and_parse_log_futures, parse_logs_in_range, parse_logs_in_tx,
-    CONTRACT_ADDRESS_ATTRIBUTE_KEY, CONTRACT_ADDRESS_ATTRIBUTE_KEY_BASE64,
-};
-use crate::{ConnectionConf, CosmosMailbox, HyperlaneCosmosError, Signer};
+use crate::indexer::{CosmosEventIndexer, ParsedEvent};
+use crate::utils::{CONTRACT_ADDRESS_ATTRIBUTE_KEY, CONTRACT_ADDRESS_ATTRIBUTE_KEY_BASE64};
+use crate::{cw::CwQueryClient, CosmosAddress, CosmosProvider, HyperlaneCosmosError, RpcProvider};
+
+use super::CwMailbox;
 
 /// The message dispatch event type from the CW contract.
-pub const MESSAGE_DISPATCH_EVENT_TYPE: &str = "mailbox_dispatch";
+pub const MESSAGE_DISPATCH_EVENT_TYPE: &str = "wasm-mailbox_dispatch";
```

Comment:
Same question is [here](https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/6012/files#r2059999080)

Reply:
Resolving with this - https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/6012#discussion_r2060018613